mobile/src/helpers/health.ts

```diff
@@ -2,14 +2,138 @@ import GoogleFit, { Scopes } from 'react-native-google-fit';
 import AppleHealthKit from 'react-native-health';
 import logger from 'common/logger';
 
+// GOOGLE FIT:
 const runOptions = {
   scopes: [
     Scopes.FITNESS_ACTIVITY_READ,
-    Scopes.FITNESS_ACTIVITY_WRITE,
-    Scopes.FITNESS_BODY_READ,
-  ]
+    Scopes.FITNESS_SLEEP_READ
+  ],
 };
 
+//Google Fit Steps
+let dailyStepCount;
+
+var today = new Date();
+var lastWeekDate = new Date(
+  today.getFullYear(),
+  today.getMonth(),
+  today.getDate() - 8,
+);
+const opt = {
+  startDate: lastWeekDate.toISOString(), // required ISO8601Timestamp
+  endDate: today.toISOString(), // required ISO8601Timestamp
+  bucketUnit: 'DAY', // optional - default "DAY". Valid values: "NANOSECOND" | "MICROSECOND" | "MILLISECOND" | "SECOND" | "MINUTE" | "HOUR" | "DAY"
+  bucketInterval: 1, // optional - default 1.
+};
+
+const fetchStepsData = async opt => {
+  const res = await GoogleFit.getDailyStepCountSamples(opt);
+  if (res.length !== 0) {
+    for (var i = 0; i < res.length; i++) {
+      if (res[i].source === 'com.google.android.gms:estimated_steps') {
+        let data = res[i].steps.reverse();
+        dailyStepCount = res[i].steps;
+        // setdailySteps(data[0].value);
+        return dailyStepCount;
+      }
+    }
+  } else {
+    logger.log('Daily Steps Not Found');
+  }
+};
+
+const fetchSleepData = async opt => {
+  const midnight = "00:00"
+  // var midnight = new Date();
+  // midnight.setHours(0, 0, 0, 0);
+  let sleepTotal = 0;
+  const res = await GoogleFit.getSleepSamples(opt);
+
+  for (var i = 0; i < res.length; i++) {
+    if (Date.parse(res[i].endDate) > Date.parse(midnight)) {
+      if (Date.parse(res[i].startDate) > Date.parse(midnight)) {
+        sleepTotal +=
+          Date.parse(res[i].endDate) - Date.parse(res[i].startDate);
+      } else {
+        sleepTotal += Date.parse(res[i].endDate) - Date.parse(midnight);
+      }
+      if (
+        i + 1 < res.length &&
+        Date.parse(res[i].startDate) < Date.parse(res[i + 1].endDate)
+      ) {
+        sleepTotal -=
+          Date.parse(res[i + 1].endDate) - Date.parse(res[i].startDate);
+      }
+    }
+  }
+  return Math.round((sleepTotal / (1000 * 60 * 60)) * 100) / 100;
+  // setSleep(Math.round((sleepTotal / (1000 * 60 * 60)) * 100) / 100);
+};
+
+const fetchAndroidHealthData = async () => {
+      let sleepData;
+      let stepsData;
+      // call Google Fit API to get the steps data for the user
+      // opts --> options object
+      logger.log('GOOGLE_FIT: Fetching Step Data');
+      stepsData = await fetchStepsData(opt);
+
+      // call Google Fit API to get the sleep data for the user
+      // opts --> options object
+      logger.log('GOOGLE_FIT: Fetching Sleep Data');
+      sleepData = await fetchSleepData(opt);
+
+      logger.log('GOOGLE_FIT: Sleep Data: ', JSON.stringify(sleepData));
+      logger.log('GOOGLE_FIT: Step Data: ', JSON.stringify(stepsData));
```

Comment:
There are a lot of logger and console.log statements in this file, remove all except the most useful for future devs